demos/unity_states/unity/Assets/Scripts/BallStateDataSourceSynchronizer.cs

```diff
@@ -0,0 +1,49 @@
+using UnityEngine;
+using TP = Teleportal.Client;
+using States = Teleportal.Client.Contract.Properties.States;
+using RSharp;
+
+public class BallStateDataSourceSynchronizer : MonoBehaviour
+{
+    [SerializeField] private BallStateDataSourcePlatform srcPlatform;
```

Comment:
Don't these need to be retrieved as components? I don't see where they are initialized

Reply:
They are assigned in the inspector in Unity Editor, hence the attachment of the `SerializedField` attribute. Click on the `DataSourceSynchronizer` object in the test scene hierarchy.

![Screen Shot 2022-07-29 at 2 57 11 PM](https://user-images.githubusercontent.com/5103968/181848211-067c92ee-860a-473c-8661-3c8fb9fb3ad2.png)